### PR TITLE
feat(@angular/cli): add `warnings.cliDependency` option for disabling dependency warning

### LINKED ISSUE #9721: Feature Request: Add option to disable 'Unable to find "@angular/cli" in devDependencies' warning
### Versions
```
Unable to find "@angular/cli" in devDependencies.

Please take the following steps to avoid issues:
"npm install --save-dev @angular/cli@latest"

Angular CLI: 1.7.0
Node: 8.7.0
OS: darwin x64
Angular: 5.2.5
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.7.0
@angular-devkit/build-optimizer: 0.3.1
@angular-devkit/core: 0.3.1
@angular-devkit/schematics: 0.3.1
@ngtools/json-schema: 1.2.0
@ngtools/webpack: 1.10.0
@schematics/angular: 0.3.1
@schematics/package-update: 0.3.1
typescript: 2.5.3
webpack: 3.11.0
```

### Use case

We're managing several angular cli based setups. For maintaining these we've created a meta package that has @angular/cli as a dependency and several other packages that our projects are dependent on. For maintenance we're centralizing our updates in the meta package so the project owner does not need to worry about these maintenance items and this lowers our cost of maintenance on multiple projects.

Sample meta package dependencies:

```json
    "@angular/animations": "^5.2.0",
    "@angular/cli": "~1.7.0",
    "@angular/common": "^5.2.0",
    "@angular/compiler": "^5.2.0",
    "@angular/compiler-cli": "^5.2.0",
    "@angular/core": "^5.2.0",
    "@angular/forms": "^5.2.0",
    "@angular/http": "^5.2.0",
    "@angular/language-service": "^5.2.0",
    "@angular/platform-browser": "^5.2.0",
    "@angular/platform-browser-dynamic": "^5.2.0",
    "@angular/router": "^5.2.0",
    "@angularclass/hmr": "^2.1.3",
    "@types/jasmine": "~2.8.3",
    "@types/jasminewd2": "~2.0.2",
    "@types/node": "~6.0.60",
    "classlist.js": "^1.1.20150312",
    "codelyzer": "^4.0.1",
    "console-polyfill": "^0.3.0",
    "core-js": "^2.4.1",
    "cz-conventional-changelog": "^1.2.0",
    "jasmine-core": "~2.8.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~2.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "rxjs": "^5.5.6",
    "standard-changelog": "~1.0.7",
    "svgxuse": "^1.2.6",
    "ts-node": "~4.1.0",
    "tslint": "~5.9.1",
    "typescript": "~2.5.3",
    "web-animations-js": "^2.3.1",
    "zone.js": "^0.8.19"
  ```

Sample project dependencies:

```json
  "dependencies": {
    "@ngrx/effects": "^5.1.0",
    "@ngrx/entity": "^5.1.0",
    "@ngrx/router-store": "^5.0.1",
    "@ngrx/store": "^5.1.0",
    "meta-package": "git+ssh://git@bitbucket.org/some-git/repo.git"
  },
  "devDependencies": {
    "@ngrx/schematics": "^5.1.0",
    "@ngrx/store-devtools": "^5.1.0",
    "karma": "~2.0.0",
    "ngrx-store-freeze": "^0.2.1"
  }
```

### Observed behavior
Running any command produces the below, but otherwise runs fine:
```
Unable to find "@angular/cli" in devDependencies.

Please take the following steps to avoid issues:
"npm install --save-dev @angular/cli@latest"
```

### Desired behavior

A way to hide this message.

### PR DESCRIPTION
**Pull request by admosity**
Closes #9721

### CODE REVIEW

File: packages/@angular/cli/bin/ng [hunk 1]
```diff
@@ -98,8 +98,9 @@ if (packageJsonProjectPath && fs.existsSync(packageJsonProjectPath)) {
   const hasDep = !!deps['@angular/cli'];
   const hasOldDevDep = !!devDeps['angular-cli'];
   const hasDevDep = !!devDeps['@angular/cli'];
+  const hasDependencyWarning = CliConfig.fromProject().get('warnings.cliDependency');
```
Comment (hansl) [inline packages/@angular/cli/bin/ng]: `fromGlobal()`, you don't have a project in `ng new`.

Comment (admosity) [reply packages/@angular/cli/bin/ng]: Thanks for the review! Updated and squashed.

Comment (admosity) [reply packages/@angular/cli/bin/ng]: Hmm, looks like I'll want to read from the project configuration if it is within a project. I'll make that change.

Comment (admosity) [reply packages/@angular/cli/bin/ng]: With and without pulling in the project config:
image

Looks like `fromProject()` returns `null` when it's not available so I just have a conditional to swap between the two.

File: packages/@angular/cli/bin/ng [hunk 2]
```diff
@@ -98,8 +98,11 @@ if (packageJsonProjectPath && fs.existsSync(packageJsonProjectPath)) {
...
   const hasOldDevDep = !!devDeps['angular-cli'];
   const hasDevDep = !!devDeps['@angular/cli'];
+  // prioritize pulling from project configuration if available otherwise fallback to global
+  const baseConfiguration = CliConfig.fromProject() || CliConfig.fromGlobal();
+  const hasDependencyWarning = baseConfiguration.get('warnings.cliDependency');
 
-  if (hasOldDep || hasOldDevDep || !(hasDevDep || hasDep)) {
+  if (hasDependencyWarning && ( hasOldDep || hasOldDevDep || !(hasDevDep || hasDep) )) {
     const warnings = [
       'Unable to find "@angular/cli" in devDependencies.',
```
Comment (clydin) [inline packages/@angular/cli/bin/ng]: Should the message be updated to reflect that this actually checks both `dependencies` and `devDepedencies`?

Comment (admosity) [reply packages/@angular/cli/bin/ng]: Looks to be the case. Should I make the change?
'Unable to find "@angular/cli" in devDependencies.' => 'Unable to find "@angular/cli" in devDependencies or dependencies.'

Comment (clydin) [reply packages/@angular/cli/bin/ng]: might as well while this section is being updated

Comment (admosity) [reply packages/@angular/cli/bin/ng]: Alrighty updated 👍

**Comment by clydin** [comment]
Closing as this warning has been removed.

**Pull request closed without merging**